src/main/resources/META-INF/resources/primefaces-extensions/badge/badge.js

```diff
@@ -18,42 +18,20 @@ PrimeFaces.widget.ExtBadge = PrimeFaces.widget.BaseWidget.extend({
         this.cfg = cfg;
         this.target = PrimeFaces.expressions.SearchExpressionFacade.resolveComponentsAsSelector(this.cfg.target);
 
-        var button = this.target;
-        if (this.target.is('.ui-button')) {
-            button = this.target;
-        } else {
-            button = $(':button:enabled:first', this.target);
```

Comment:
@melloware I don't get what this line was doing? Can you please explain?

Reply:
The the target was NOT a button say you just had Badge inside a panel it would look for the FIRST enabled button within that component to attach to.